mmseg/core/evaluation/metrics.py

```diff
@@ -280,16 +280,28 @@ def eval_metrics(results,
         ndarray: Per category accuracy, shape (num_classes, ).
         ndarray: Per category evaluation metrics, shape (num_classes, ).
     """
-
-    total_area_intersect, total_area_union, total_area_pred_label, \
-        total_area_label = total_intersect_and_union(
-            results, gt_seg_maps, num_classes, ignore_index, label_map,
-            reduce_zero_label)
-    ret_metrics = total_area_to_metrics(total_area_intersect, total_area_union,
-                                        total_area_pred_label,
-                                        total_area_label, metrics, nan_to_num,
-                                        beta)
-
+    if isinstance(metrics, str):
+        metrics = [metrics]
+    seg_pred_metrics = ['mIoU', 'mDice', 'mFscore']
+    if set(metrics).issubset(set(seg_pred_metrics)):
+        results = results['seg_pred'] if isinstance(results, dict) else results
+        total_area_intersect, total_area_union, total_area_pred_label, \
+            total_area_label = total_intersect_and_union(
+                results,
+                gt_seg_maps,
+                num_classes,
+                ignore_index,
+                label_map,
+                reduce_zero_label,
+            )
+        ret_metrics = total_area_to_metrics(total_area_intersect,
+                                            total_area_union,
+                                            total_area_pred_label,
+                                            total_area_label, metrics,
+                                            nan_to_num, beta)
+    elif metrics == 'xxx':
+        # place holder for additional metric
+        pass
```

Comment:
```suggestion
```

Reply:
Maybe making 302-304 as comments is better?